functions/Get-DbaInstallDate.ps1

```diff
@@ -0,0 +1,183 @@
+		[Switch]$SqlOnly,
+		[parameter(ParameterSetName = "Windows")]
+		[Switch]$WindowsOnly,
+		[Alias("Credential")]
+		[switch]$Silent
+	)
+	
+	PROCESS
+	{
+		foreach ($instance in $SqlServer)
+		{
+			if ($instance.Gettype().FullName -eq [System.Management.Automation.PSCustomObject] )
+			{
+				$servername = $instance.SqlInstance
+			}
+			elseif ($instance.Gettype().FullName -eq [Microsoft.SqlServer.Management.Smo.Server])
+			{
+				$servername = $instance.NetName
+			}
+			else
+			{
+				$servername = $instance
+			}
+						
+			if ($WindowsOnly -ne $true)
+			{ 
+				try 
+				{
+					Write-Message -Level Verbose -Message "Connecting to $instance"
+                    if ($SqlCredential)
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername -SqlCredential $SqlCredential
+                    } else 
+                    { 
+                        $server = Connect-SqlInstance -SqlInstance $servername 
+                    }					
+				}
+				catch 
+				{
+					Stop-Function -Message "Failed to connect to: $instance" -Continue -Target $instance					
+				}
+
+				if ( $server.VersionMajor -ge 9 )
+				{ 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" 
+					$sql = "SELECT create_date FROM sys.server_principals WHERE sid = 0x010100000000000512000000"
+					$sqlInstallDate = $server.query('master',$sql).tables[0]
+					[SqlCollective.dbatools.Utility.DbaDateTime]$sqlInstallDate = $sqlInstallDate.Rows.Create_Date
+
+				} else { 
+					Write-Message -Level Verbose -Message "Getting Install Date for: $instance" 
+					$sql = "SELECT schemadate FROM sysservers"
+					$sqlInstallDate = $server.query('master',$sql).tables[0]
+					[SqlCollective.dbatools.Utility.DbaDateTime]$sqlInstallDate = $sqlInstallDate.Rows.Create_Date
+				}											
+
+			} 
+
+			if ($SqlOnly -ne $true)			
+			{ 
+				Write-Message -Level Verbose -Message "Getting Windows Server Name for: $servername" 
+				$WindowsServerName = (Resolve-DbaNetworkName $servername -Credential $WindowsCredential).ComputerName
+				
+				try
+				{
+					Write-Message -Level Verbose -Message "Getting Windows Install date via CIM for: $WindowsServerName"
+					$WindowsServerName		
+					[SqlCollective.dbatools.Utility.DbaDateTime]$windowsInstallDate = (Get-CimInstance -ClassName win32_operatingsystem -ComputerName $windowsServerName -ErrorAction SilentlyContinue).InstallDate								
+				}
+				catch
+				{
+					try
+					{	
+						Write-Message -Level Verbose -Message "Getting Windows Install date via DCOM for: $WindowsServerName" 
+						$CimOption = New-CimSessionOption -Protocol DCOM
+						$CimSession = New-CimSession -Credential:$WindowsCredential -ComputerName $WindowsServerName -SessionOption $CimOption
+						[SqlCollective.dbatools.Utility.DbaDateTime]$windowsInstallDate = ($CimSession | Get-CimInstance -ClassName Win32_OperatingSystem).LastBootUpTime					
+					}
```

Comment:
So playing with the `Get-DbaCmObject` you should be able to take line 149 through 151 and just do this:

```powershell
[DbaDateTime]$windowsInstallDate = (Get-DbaCmObject -ComputerName $WindowsServerName -ClassName win32_OperatingSystem -Credential $WindowsCredential).LastBootUpTime
```

I don't have domain computer and remote one to test this one but that should work as I understand the function so far.